src/query/sql/src/planner/semantic/type_check.rs

```diff
@@ -1805,10 +1817,14 @@ impl<'a> TypeChecker<'a> {
 
             let accessor = accessores.pop().unwrap();
             let accessor_lit = match accessor {
-                MapAccessor::Bracket { key } => key.clone(),
+                MapAccessor::Bracket {
+                    key: box Expr::Literal { lit, .. },
+                } => lit,
                 MapAccessor::Period { key } | MapAccessor::Colon { key } => {
                     Literal::String(key.name.clone())
                 }
+                MapAccessor::PeriodNumber { key } => Literal::Integer(key - 1),
```

Comment:
The inner name of the tuple field is starting from 0, while the index of `<tuple>.<index>` in the surface SQL is starting from 1. Should we change the inner name for consistency?